content/topics/dotenv-module/_index.md

```diff
@@ -0,0 +1,183 @@
+---
+title: Introduction to Dotenv module 
+ready: true
+
+---
+
+## What is an environment variable?
+
+An [environment variable](https://en.wikipedia.org/wiki/Environment_variable) is a variable whose value is set outside the program, typically through functionality built into the operating system or [container](https://www.docker.com/resources/what-container). An environment variable is made up of a name/value pair, and any number may be created and available for reference at a point in time.
+
+For a more in-depth description of environmental variables, take a look at this: {{% contentlink "topics/linux/os-environmental-variables/" %}}
+
+```
+//example 
+NAME_OF_PLACE=Umuzi
+
+```
+
+## How does dotenv help you?
+
+Using environment variables is one technique to make your app easier to configure by separating infrequently changing data from your code. A strict separation of config from code means we can change config (which there's a small amount of, and it doesn't need to take much time) without needing to change any code and recompile (which can be big and slow). This makes for a portable app as you can have different configurations for multiple scenarios i.e configs for ```Development mode``` and ```Production mode``` without changing a single line in your code. [Dotenv](https://www.npmjs.com/package/dotenv) module loads environment variables from a .env file that you create and adds them to the [process.env](https://nodejs.org/docs/latest/api/process.html#process_process_env) object that is made available to the application
+
```

Comment:
This is going to confuse the noobs. This proves that you know how it works but that's not really the point of this document.